protocol/0014-ORDT-order_types.md

```diff
@@ -90,6 +90,76 @@ Network orders are used during [position resolution](./0012-POSR-position_resolu
+6. An iceberg order with minimum peak size less than 0 is rejected with a valid error message.
+7. An iceberg order with minimum peak size greater than initial peak size is rejected with a valid error message.
+
+### Iceberg Order Amendment
+
+1. Amending an iceberg order to increase size will increase the total and remaining quantities of the order and time priority of the order is not lost.
+2. Amending an iceberg order to decrease size will decrease the total and remaining quantities and time priority of the order is not lost.
+3. Amend an iceberg order to decrease size so that the displayed quantity is decreased. Total and remaining quantity is decreased, margin is recalculated and released and time priority is not lost.
+4. What other scenarios do we need to consider here ?
+
+### Iceberg Order Cancellation
+
+1. Cancelling an iceberg order will cancel the order, remove it from the order book , release margin and update order book to reflect the change.
+
+### Iceberg Order Execution
+
+1. An aggressive iceberg order that crosses with an order where volume > iceberg volume, the iceberg order gets fully filled on entry, the iceberg order status is filled, the remaining quantity = 0. Are atomic trades generated OR one single trade for each display quantity volume ???
+2. An aggressive iceberg order that crosses with an order where volume < iceberg volume. The initial display quantity is filled and the remaining volume is unfilled. Status of iceberg order is partially filled , the volume remaining = (quantity - initial volume) and the remaining volume sits on the book. When additional orders thar are submitted which consume the remaining volume on the iceberg order , the volume of the iceberg order is refreshed as and when the volume dips below the minimum peak size.
+3. A passive iceberg order (the only order at a particular price level) when crossed with another order that comes in which consumes the full volume of the iceberg order is fully filled. Status of iceberg order is filled and the remaining = 0. Atomic trades are produced.
+4. A passive iceberg order with a couple of order that sit behind the iceberg order at the same price that crosses with an order where volume > display quantity of iceberg order. After the first trade is produced , the iceberg order is pushed to the back of the queue and gets filled only when the other orders in front get fully filled.
+5. Submit an aggressive iceberg order for say size 100. There are multiple matching orders of size 30,40,50. Ensure the orders are matched and filled in time priority of the orders and any remaining volume on the orders is correctly left behind.
+6. Submit an aggressive iceberg order for say size 100. There are multiple matching orders of size 20,30. Ensure the orders are matched and filled in time priority of the orders. Ensure remaining volume on the iceberg order is (100 - (20+30))
```

Comment:
These two may not be worth worrying about, the general matching engine should probably be handling time priority fine already

Reply:
Thanks , have marked them as low priority.